plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DataFileInfo.java

```diff
@@ -28,20 +28,23 @@
     private final long size;
     private final long creationTime;
     private final DeltaLakeJsonFileStatistics statistics;
+    private final boolean isCDFData;
 
     @JsonCreator
     public DataFileInfo(
             @JsonProperty("path") String path,
             @JsonProperty("size") long size,
             @JsonProperty("creationTime") long creationTime,
             @JsonProperty("partitionValues") List<String> partitionValues,
```

Comment:
Also here - I'm not sure whether it is relevant - do consider having different classes for normal delta lake files and change data files.

Reply:
I considered that but it would require a lot of changes in how we propagate information about written files, how we write them and clean them, if necessary i can add this later as a separate commit but imho it is not necessary.

Reply:
It is also used in https://github.com/trinodb/trino/blob/a18e5e96f584fb443d195ccd48e1d6564eb6c986/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeWriter.java#L169 and through it it is used as a result from PageSinks

Reply:
I tried on my own to split the classes:

- `DataFileInfo` extends `FileInfo`
- `CdfFileInfo` extends `FileInfo`

and stumbled by chance on a few inconsistencies.


`DeltaLakeWriter` creates statistics (even though not needed) for CDF `DataFileInfo` instances.
The `DataFileInfo` instances currently with `cdfData` set to `false` created in the `DeltaLakeWriter` are being translated to `DataFileInfo` instances with `cdfData` set to `true` within `io.trino.plugin.deltalake.DeltaLakeCDFPageSink#buildResult` - this is a pragmatic choice which makes things a bit hard to follow.

If the split is being taken into consideration, I'd advocate for a `DeltaLakeCdfWriter` class (to be used in `DeltaLakeCdfPageSink`) specially dedicated for writing CDF files which avoids calculating stats and dealing with `timestampColumnIndices`.


On the other hand, you are right in saying that there are a lot of changes to be performed if we separate `DataFileInfo` in two independent classes.

Reply:
I'd prefer to do it as a separate PR, it looks like an optimization

Reply:
In case the maintainer will feel the same way, could you please create a follow-up issue to get this aspect eventually addressed?

Reply:
> Or we could add a `ContentType` enum, with values of `DATA` or `CDF`

sounds simple

> , I'd advocate for a DeltaLakeCdfWriter class (to be used in DeltaLakeCdfPageSink) specially dedicated for writing CDF files which avoids calculating stats and dealing with timestampColumnIndices.

that won't make the code simpler, will it?

that can make the code more performant. do we have idea by how much?

Reply:
> that can make the code more performant. do we have idea by how much?

I don't know HOW much.

I would say that the improvement would be of the duration:
```
count(CDF files) x analyze stats
```